make a bunch of app_manager js use hqDefine

dannyroberts opened the pull request:
You'll definitely want to review this ignoring whitespace (https://github.com/dimagi/commcare-hq/pull/10503/files?w=1)

dannyroberts committed:
move detail-screen-config.js over to hqDefine

dimagimon commented on corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js:
```diff
@@ -14,7 +18,7 @@ var CC_DETAIL_SCREEN = {
         if (parts[j][0] == '#') {
             parts[j] = ('<span class="label label-info">'
-                        + CC_DETAIL_SCREEN.toTitleCase(parts[j]) + '</span>');
+                        + module.CC_DETAIL_SCREEN.toTitleCase(parts[j]) + '</span>');
```
Bad line breaking before '+'.

dannyroberts committed:
move app_manager_media.js over to hqDefine

dannyroberts committed:
Merge branch 'master' into use-hqImport

Conflicts:
	corehq/apps/app_manager/templates/app_manager/partials/case_list_properties.html

dannyroberts committed:
make app_manager_utils use hqDefine

dannyroberts committed:
refactor nav_menu_media_js.html

to make it a thin wrapper around a pure js function
separating js from django templates

dannyroberts committed:
make careplan-config-ui.js use hqDefine

dannyroberts committed:
make case-config-ui-2.js use hqDefine

dannyroberts committed:
make case-config-ui-advanced.js use hqDefine

dannyroberts committed:
make case-config-utils.js use hqDefine

dannyroberts committed:
make case-knockout-bindings.js leave no global footprint

other than modifying ko.bindingHandlers of course

dannyroberts committed:
make commcaresettings.js use hqDefine

dannyroberts committed:
fix indentation of detail-screen-config.js module

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts committed:
make download_async_modal.js use hqDefine

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts committed:
make form_workflow.js use hqDefine

dannyroberts committed:
add hqModules to mocha test html file

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts committed:
make graph-config.js use hqDefine

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts committed:
make releases.js use hqDefine

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts committed:
make report-module.js use hqDefine

dannyroberts committed:
make supported-languages.js use hqDefine

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts committed:
make visit-scheduler.js use hqDefine

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts commented:
@orangejenny cc: @kaapstorm buddy

orangejenny commented on corehq/apps/app_manager/static/app_manager/js/app_manager_media.js:
```diff
@@ -1,150 +1,190 @@
-                self.customPath(data.ref.path);
-            }
+    function initNavMenuMedia(qualifier, imageRef, audioRef, objectMap, defaultFileName) {
+        var uploaders = hqImport('#app_manager/partials/nav_menu_media_js_common.html');
```
Why the leading `#`?

dannyroberts replied:
I guess because this isn't actually a static file, I felt like I should identify that with a different convention. I saw in your PR you had a similar convention with `#moduleName` at the end of the path.

dannyroberts replied:
I'm not in love with or married to it, to use romantic metaphors

orangejenny replied:
Cool, makes sense. I think we should pick one style (`#modulename/file/name.html` or `file/name.html#modulename` or whatever)...but I'm about to kill keyboard shortcuts, so right now I think this is the only place defining a module in an HTML file.

millerdev replied:
Ignore me if I am too late for this party. My vote would be for `file/name.html#modulename` since that that allows us to intuitively define multiple modules per HTML file (each script block could potentially have its own module definition).

orangejenny commented:
Looks great! Left one minor question.

kaapstorm commented:
:+1:  hqDefine and hqImport look really good in use.

dannyroberts committed:
improve bitly_nato_phonetic comment

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

orangejenny commented:
Good to merge once QA passes.

dannyroberts committed:
Merge branch 'master' into use-hqImport

Conflicts:
	corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts committed:
Merge branch 'master' into use-hqImport

Conflicts:
	corehq/apps/app_manager/static/app_manager/js/releases.js

dannyroberts commented:
I poked around on staging and have tested this a few different times, and I'm pretty sure it doesn't break anything. In order for this to get deployed, we have to merge https://github.com/dimagi/commcare-hq/pull/10643 first (or else the js files won't cachebust).

I spent most of today fixing things related to this PR working on staging: conflicts with other branches, this static files thing, etc. And I'm pretty confident in it :)

@dimagi/js-team

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

millerdev commented:
can't merge

dannyroberts committed:
Merge branch 'master' into use-hqImport

Conflicts:
	corehq/apps/app_manager/static/app_manager/js/releases.js

dimagimon commented:
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js, line 17 - 'MAXLEN' is already defined.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 45 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 47 - Unnecessary semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 100 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 165 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 166 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 183 - Use '!==' to compare with 'null'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 261 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 339 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 384 - Confusing use of '!'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 399 - ['values'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 400 - ['value_names'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['hq'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 410 - ['application_version'] is better written in dot notation.

dannyroberts commented:
Thanks, merged and did a quick click pass around app builder locally.

dannyroberts commented:
:white_check_mark:

dannyroberts commented:
Thanks @millerdev, so glad to finally get this in! It was causing almost daily (if simple) merge conflicts.
